docs/order_of_PCI_addresses.md: add info about PCI addresses (#182)
pcengines/apu2-documentation: docs/order_of_PCI_addresses.md

Order of PCI addresses
===============================

## Standard PCI addressing

> Coreboots standard way of numbering PCI devices prioritizes PCIe slots over

Reviewer: coreboot always small letter

> embedded PCI devices like Ethernet interfaces. It causes weird and unwanted

Reviewer: This is not exactly true. coreboot doesn't do anything wrong. It was the developer who initially wrote the support for apu2 and defined the PCIe engines in weird order to be initialized by AGESA. What coreboot does it to enumerate the PCI devices in ascending order as it should. I would remove this section and write it from scratch.

Write about:
- bad PCIe engines definition passed to AGESA in the initial coreboot port  for apu2 resulting in impractical order of PCIe devices

behavior like shifting addresses of old devices when a new one is added.
Unfortunately it can’t be changed in APU without consequences, because of many
years of it existence on the market. If it had been changed, many APU owners
would be surprised by e.g. change of interfaces names.

## sortbootorder

To address this issue, we added a runtime option in [sortbootorder](https://github.com/pcengines/sortbootorder/blob/master/README.md)

> which can reverse those priorities. By setting `Reverse order of PCI addresses`

Reviewer: I would not use priorities, because that makes no sense. Write something like this:
```
Reverses the order of PCIe engines assignment in a way that:
- NICs occupy the lowest PCIe device numbers which make their enumeration persistent regardless of the WiFi module presence
- the WoL capable NIC is the first PCIe device in iPXE and in the operating system
- the mPCIe slots occupy the highest PCIe device numbers in following order: 3x NIC, mPCIe1, mPCIe2. In such case faulty detected WiFi in mPCIe2 slot will not affect interface renaming for mPCIe1 slot WiFi card.

By default, this option is disabled which means nothing changes (compared to the state maintained for past 4 years) without intentional change of this setting.
```

option in sortbootorder menu it is possible to force firmware to number PCI
devices in the way that newly added devices don't take places of old ones, thus
fixing possible issues like unwanted changes of interfaces names. By default
this option is disabled which means nothing changes without intentional change
of this setting.

## How to change

> To reverse order of PCI addresses, firstly enter the sortbootorders menu in

Reviewer: ```suggestion
To reverse order of PCI addresses, enter the sortbootorder first in the
```

> SeaBios menu while booting by choosing `payload [setup]`, then enable the

Reviewer: ```suggestion
SeaBIOS menu while booting by choosing `payload [setup]`, then enable the
```

Order of PCI addresses
===============================

## Standard PCI addressing

Initial coreboot port for APU2 was made with badly implemented PCIe engines
definition passed to AGESA. It's done in the way that mPCIe cards get lower

> address then Network interface controllers (NICs). It causes weird and unwanted

Reviewer: ```suggestion
address than Network interface controllers (NICs). It causes weird and unwanted
```
